core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -518,16 +516,19 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
     }
 
     // Validate spark.executor.extraJavaOptions
-    getOption(executorOptsKey).foreach { javaOpts =>
-      if (javaOpts.contains("-Dspark")) {
-        val msg = s"$executorOptsKey is not allowed to set Spark options (was '$javaOpts'). " +
-          "Set them directly on a SparkConf or in a properties file when using ./bin/spark-submit."
-        throw new Exception(msg)
-      }
-      if (javaOpts.contains("-Xmx")) {
-        val msg = s"$executorOptsKey is not allowed to specify max heap memory settings " +
-          s"(was '$javaOpts'). Use spark.executor.memory instead."
-        throw new Exception(msg)
+    Seq(EXECUTOR_JAVA_OPTIONS.key, "spark.executor.defaultJavaOptions").foreach { executorOptsKey =>
```

Comment:
Is there not a constant already for `spark.executor.defaultJavaOptions`? not a big deal

Reply:
Constants are defined in the `SparkLauncher` class, I avoid import `SparkLauncher` in `SparkConf`.

https://github.com/apache/spark/blob/cf64008fce77b38d1237874b04f5ac124b01b3a8/launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java#L71